docs/userguide.md

```diff
@@ -387,3 +387,164 @@ employed by deployment:
+to actually propagate the k8s resource to federated clusters. Its noteworthy that 
+if an RSP is present, the `spec.replicas` from the `template` resource are unused. 
+RSP also provides a further more useful feature using `spec.rebalance`. If this is 
+set to `true`, the RSP controller monitors the replica pods for target replica 
+workload from each federated cluster and if it finds that some clusters are not 
+able to schedule those pods for long, it moves (rebalances) the replicas to 
+clusters where all the pods are running and healthy. This in other words helps 
+moving the replica workloads to those clusters where there is enough capacity 
+and away from those clusters which are currently running out of capacity. The 
+`rebalance` feature might cause initial shuffle of replicas to  reach an eventually 
+balanced state of distribution. The controller might further keep trying to move 
+few replicas back into the cluster(s) which ran out of capacity, to check if it can 
+be scheduled again to reach the normalised state (even distribution or the state 
+desired by user preferences), which apparently is the only mechanism to check if 
+this cluster has capacity now. The `spec.rebalance` should not be used if this 
+behaviour is unacceptable.
+
+The RSP can be considered as more user friendly mechanism to distribute the 
+replicas, where the inputs needed from the user at federated control plane are 
+reduced. The user only needs to create the RSP resource and the mapping template 
+resource, to distribute the replicas. It can also be considered as a more 
+automated approach at distribution and further reconciliation of the workload 
+replicas.
+
+The usage of the RSP semantics is illustrated using some examples below. The 
+examples considers 3 federated clusters `A`, `B` and `C`.
+
+#### Distribute total replicas evenly in all available clusters
+
+```bash
+apiVersion: scheduling.federation.k8s.io/v1alpha1
+kind: ReplicaSchedulingPreference
+metadata:
+  name: test-deployment
+  namespace: test-ns
+spec:
+  targetKind: FederatedDeployment
+  totalReplicas: 9
+```
+or 
+```bash
+apiVersion: scheduling.federation.k8s.io/v1alpha1
+kind: ReplicaSchedulingPreference
+metadata:
+  name: test-deployment
+  namespace: test-ns
+spec:
+  targetKind: FederatedDeployment
+  totalReplicas: 9
+  clusters:
+    "*":
+      weight: 1
+```
+A, B and C get 3 replicas each.
+
+#### Distribute total replicas in weighted proportions
+
+```bash
+apiVersion: scheduling.federation.k8s.io/v1alpha1
+kind: ReplicaSchedulingPreference
+metadata:
+  name: test-deployment
+  namespace: test-ns
+spec:
+  targetKind: FederatedDeployment
+  totalReplicas: 9
+  clusters:
+    A:
+      weight: 1
+    B:
+      weight: 2
+```
+A gets 3 and B gets 6 replicas in the proportion of 1:2. C does not get 
+any replica as missing weight preference is considered as weight=0.
+
+#### Distribute replicas in weighted proportions, also enforcing replica limits per cluster
+
+```bash
```

Comment:
Would you please show more detail for how does the `minReplicas` will impact the placement policy? I created a policy as this:

```
apiVersion: scheduling.federation.k8s.io/v1alpha1
kind: ReplicaSchedulingPreference
metadata:
  name: test-deployment
  namespace: test-namespace
spec:
  targetKind: FederatedDeployment
  totalReplicas: 10
  rebalance: true
  clusters:
   cluster1:
   - minReplicas: 2
     maxReplicas: 6
     weight: 2
   cluster2:
   - minReplicas: 4
     maxReplicas: 8
     weight: 3
```

And the result is cluster1 has 3 and cluster2 has 7, and my expectation is cluster1 should have 4 while cluster2 should have 6.

Reply:
Yes, that looks like a bug.  Your expectation is the same as mine.  Please file an issue.

Reply:
Thanks @quinton-hoole-2 , opened an issue at https://github.com/kubernetes-sigs/federation-v2/issues/319 to trace.

Reply:
I think you have already raised it as an issue, lets track this as a bug there.